wagtail/models/sites.py

```diff
@@ -15,6 +17,8 @@
 MATCH_DEFAULT = 2
 MATCH_HOSTNAME = 3
 
+_root_paths_cache = Local()
```

Comment:
We could technically use this for other things, such as the request hostname/port and the current site (lazily determined from the former), but I think this name makes sense for now